--- proj.git/src/log.cpp	2022-02-23 19:51:04.000000000 +0100
+++ proj.git/src/log.cpp	2022-06-16 20:47:17.420427109 +0200
@@ -37,12 +37,19 @@
 /*                          pj_stderr_logger()                          */
 /************************************************************************/
 
-void pj_stderr_logger( void *app_data, int level, const char *msg )
+#ifdef ON_ANDROID
+#include <android/log.h>
+#endif
 
+void pj_stderr_logger( void *app_data, int level, const char *msg )
 {
     (void) app_data;
     (void) level;
+#ifdef ON_ANDROID
+    __android_log_print(ANDROID_LOG_INFO, "PROJ", msg);
+#else
     fprintf( stderr, "%s\n", msg );
+#endif
 }
 
 /************************************************************************/
--- proj.git/src/lib_proj.cmake	2022-02-23 19:51:04.000000000 +0100
+++ proj.git/src/lib_proj.cmake	2022-06-16 20:51:42.472055007 +0200
@@ -444,6 +444,10 @@
   target_link_libraries(proj PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 endif()
 
+if(ANDROID)
+  target_link_libraries(proj PRIVATE -llog)
+endif(ANDROID)
+
 target_include_directories(proj PRIVATE ${SQLITE3_INCLUDE_DIR})
 target_link_libraries(proj PRIVATE ${SQLITE3_LIBRARY})
 
